controllers/util/provider.go

```diff
@@ -64,16 +73,28 @@ func GenerateOnboardingToken(tokenLifetimeInHours int, privateKeyPath string, st
 	return fmt.Sprintf("%s.%s", encodedPayload, encodedSignature), nil
 }
 
-func readAndDecodePrivateKey(privateKeyPath string) (*rsa.PrivateKey, error) {
-	pemString, err := os.ReadFile(privateKeyPath)
+func decodePemKey(cl client.Client) (*rsa.PrivateKey, error) {
+	klog.Info("Decoding the Pem key")
+	ctx := context.Background()
+	operatorNamespace, err := GetOperatorNamespace()
+	if err != nil {
+		return nil, fmt.Errorf("unable to get operator namespace: %v", err)
+	}
+
+	privateSecret := &corev1.Secret{}
+	privateSecret.Name = onboardingValidationPrivateKeySecretName
+	privateSecret.Namespace = operatorNamespace
+
+	err = cl.Get(ctx, types.NamespacedName{Name: onboardingValidationPrivateKeySecretName, Namespace: operatorNamespace}, privateSecret)
```

Comment:
you have already set ObjectMeta on privateSecret and could do
```suggestion
	err = cl.Get(ctx, client.ObjectKeyFromObject(privateSecret), privateSecret)
```

Reply:
Ack